### PR TITLE
Patch 2

### PR DESCRIPTION
**Pull request by odysseusmax**

### CODE REVIEW

File: tgfilestream/web_routes.py [hunk 1]
```diff
@@ -63,8 +63,21 @@ async def handle_request(req: web.Request, head: bool = False) -> web.Response:
...
 
     size = message.file.size
-    offset = req.http_range.start or 0
-    limit = req.http_range.stop or size
+    try:
+        offset = req.http_range.start or 0
+        limit = req.http_range.stop or size
+        if (limit > size) or (offset < 0) or (limit < offset):
+            raise ValueError("range not in acceptable format")
+    except ValueError:
```
Comment (tulir) [inline tgfilestream/web_routes.py:71]: Why is this not just an `if`?

File: tgfilestream/web_routes.py [hunk 2]
```diff
@@ -63,8 +63,21 @@ async def handle_request(req: web.Request, head: bool = False) -> web.Response:
...
+        return web.Response(
+            status=416,
+            text="416: Range Not Satisfiable",
+            headers = {
+                "Content-Range": f"bytes */{size}"
+            }
+        )
+    
+    log.info(f"Range {offset} - {limit}")
```
Comment (tulir) [inline tgfilestream/web_routes.py]: This shouldn't be a separate log line, could maybe be integrated into the existing "Serving file" log

File: tgfilestream/web_routes.py [hunk 3]
```diff
@@ -63,8 +63,21 @@ async def handle_request(req: web.Request, head: bool = False) -> web.Response:
...
+        if (limit > size) or (offset < 0) or (limit < offset):
+            raise ValueError("range not in acceptable format")
+    except ValueError:
+        return web.Response(
+            status=416,
+            text="416: Range Not Satisfiable",
+            headers = {
+                "Content-Range": f"bytes */{size}"
+            }
+        )
```
Comment (tulir) [inline tgfilestream/web_routes.py]:
```suggestion
        return web.Response(status=416, text="416: Range Not Satisfiable",
                            headers={"Content-Range": f"bytes */{size}"})
```

**Review by tulir — CHANGES_REQUESTED** [review]

File: tgfilestream/web_routes.py (hunk 1 above)
Comment (odysseusmax) [reply tgfilestream/web_routes.py:71]: If http range is not correct, aiohttp raises value error. Also raising value error if limit is greater than size. Thus we can catch both errors in one try.

**Pull request merged**